demos/custom-dialogs/custom-dialog.js

```diff
@@ -0,0 +1,163 @@
+Blockly.alert = function(message, callback) {
+    console.log('Alert: ' + message);
+    CustomDialog.show('Alert', message, {
+        onCancel: callback
+    });
+};
+
+/** Override Blockly.confirm() with custom implementation. */
+Blockly.confirm = function(message, callback) {
+    console.log('Confirm: ' + message);
+    CustomDialog.show('Confirm', message, {
+        showOkay: true,
+        onOkay: function() {
+            callback(true)
+        },
+        showCancel: true,
+        onCancel: function() {
+            callback(false)
+        }
+    });
+};
+
+/** Override Blockly.prompt() with custom implementation. */
+Blockly.prompt = function(message, defaultValue, callback) {
+    console.log('Prompt: ' + message);
+    CustomDialog.show('Prompt', message, {
+        showInput: true,
+        showOkay: true,
+        onOkay: function() {
+            callback(CustomDialog.inputField.value)
+        },
+        showCancel: true,
+        onCancel: function() {
+            callback(null)
+        }
+    });
+    CustomDialog.inputField.value = defaultValue;
+};
+
+
+CustomDialog.hide = function() {
+    if (CustomDialog.backdropDiv_) {
+        CustomDialog.backdropDiv_.style.display = 'none'
+        CustomDialog.dialogDiv_.style.display = 'none'
+    }
+}
+
+/**
+ * Shows the dialog.
+ * Allowed options:
+ *  - showOkay: Whether to show the OK button.
+ *  - showCancel: Whether to show the Cancel button.
+ *  - showInput: Whether to show the text input field.
+ *  - onOkay: Callback to handle the okay button.
+ *  - onCancel: Callback to handle the cancel button.
+ */
+CustomDialog.show = function(title, message, options) {
+    var backdropDiv = CustomDialog.backdropDiv_;
+    var dialogDiv = CustomDialog.dialogDiv_;
+    if (!dialogDiv) {
+        // Generate HTML
+        var body = document.getElementsByTagName('body')[0];
+        backdropDiv = document.createElement('div');
+        backdropDiv.setAttribute('id', 'backdrop');
+        backdropDiv.style.cssText =
+              'position: absolute;'
+              + 'top: 0px;'
+              + 'left: 0px;'
+              + 'right: 0px;'
+              + 'bottom: 0px;'
+              + 'background-color: rgba(0, 0, 0, 0.7);';
+        body.appendChild(backdropDiv);
+
+        dialogDiv = document.createElement('div');
+        dialogDiv.setAttribute('id', 'dialog');
+        dialogDiv.style.cssText =
+                'background-color: white;'
+                + 'width: 400px;'
+                + 'margin: 20px auto 0;'
+                + 'padding: 10px;';
+        backdropDiv.appendChild(dialogDiv);
+
+        dialogDiv.onclick = function(event) {
+            event.stopPropagation();
+        };
+
+        CustomDialog.backdropDiv_ = backdropDiv;
+        CustomDialog.dialogDiv_ = dialogDiv;
+    }
+    backdropDiv.style.display = 'block';
+    dialogDiv.style.display = 'block';
+
+    dialogDiv.innerHTML = '<div id="dialogTitle"><strong>' + title + '</strong></div>'
```

Comment:
innerHTML is absolutely not allowed with random injections.  This is a serious security vulnerability for anyone using this code.